src/local_insight_engine/main.py

```
@@ -46,66 +68,191 @@ def analyze_document(self, document_path: Path, factual_mode: bool = False) -> d
-        document = self.document_loader.load(document_path)
-
-        # Layer 2: Process and neutralize content
-        processed_data = self.text_processor.process(document, bypass_anonymization=factual_mode)
-
-        # Layer 3: Analyze with LLM
-        analysis = self.llm_client.analyze(processed_data)
+        try:
+            # Layer 1: Load document
+            # (existing analysis logic goes here)
+            logger.step("Analysis completed successfully")
+            return analysis
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)